footprints/main/models.py

```diff
@@ -39,10 +39,16 @@ def display_format(self):
         e = EDTF(self.edtf_format)
 
         if e.is_interval:
-            return "%s - %s" % (self.fmt(e.date_obj.start),
-                                self.fmt(e.date_obj.end))
+            return "%s - %s" % (self.fmt(e.date_obj.start, True),
+                                self.fmt(e.date_obj.end, True))
         else:
-            return self.fmt(e.date_obj)
+            return self.fmt(e.date_obj, False)
+
+    def ordinal(self, n):
+        # cribbed from http://codegolf.stackexchange.com/
+        # questions/4707/outputting-ordinal-numbers-1st-2nd-3rd#answer-4712
+        return "%d%s" % (
+            n, "tsnrhtdd"[(n / 10 % 10 != 1) * (n % 10 < 4) * n % 10::4])
```

Comment:
Well, that's clever.

FWIW, if it's only going to be used for display in a template, django's included `humanize` has an ordinal filter: https://docs.djangoproject.com/en/1.9/ref/contrib/humanize/#ordinal